### PR TITLE
Add support for snappy compression over http.

### PR DESCRIPTION
**Pull request by petermattis**

### CODE REVIEW

**Comment by bdarnell** [comment]
LGTM. We should add a test that this is actually doing something: maybe manually send a request with different accept-encodings and verify that content-encoding and content-length change.

File: server/server.go [hunk 1]
```diff
@@ -181,6 +187,32 @@ func (s *Server) Stop() {
...
+// will snappy a response if the appropriate request headers are set.
+func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	// Check if we're draining; if so return 503, service unavailable.
+	if !s.stopper.StartTask() {
+		http.Error(w, "service is draining", http.StatusServiceUnavailable)
+		return
+	}
+	defer s.stopper.FinishTask()
+
+	ae := r.Header.Get("Accept-Encoding")
```
Comment (bdarnell) [inline server/server.go]: We should set `Vary: Accept-Encoding` here (or it's probably even better to disable caching completely with `Cache-control: no-cache`)

Comment (petermattis) [inline server/server.go]: Added `Cache-control: no-cache`.

**Comment by petermattis** [comment]
Enhanced the existing `TestGzip` test to also test `Accept-Encoding: snappy`.

**Comment by bdarnell** [comment]
LGTM

**Pull request merged**